src/js/module.js

```diff
@@ -126,18 +126,21 @@ iotjs_module_t.resolveFilepath = function(id, directories) {
 };
 
 
-iotjs_module_t._resolveFilepath = function(id, root) {
+iotjs_module_t._resolveFilepath = function(id, root, ext_index) {
   var modulePath = root ? path.join(root, id) : id;
   var filepath;
-  var ext = '.js';
+  var exts = ['.js', '.json', '.node'];
```

Comment:
Does Node.js searches for .json and .node? We have to consider the searching perf, a benchmark for these cases if we wanna this land :)

Reply:
Yes, Node.js searches for .json and .node. It is acceptable on performance since rest of two were searched only when .js is not found.

Reply:
Yes, I will make a benchmark.Thanks!

Reply:
Take a look at https://github.com/Rokid/ShadowNode/tree/master/benchmark/module, your benchmark should be submitted here :)

Reply:
Ok, I will do

Reply:
* To avoid the impact of caching:
  src/js/module.js:228
```js
var cachedModule = iotjs_module_t.cache[modPath];
if (false) {
  iotjs_module_t.curr = modPath;
  return cachedModule.exports;
}
```
* bench in after add `resolve` method: 
  benchmark/module/require-resolve.js
```js
'use strict';

var common = require('../common.js');

var bench = common.createBenchmark(main, {
  n: [512]
});

function main(opts) {
  var n = opts.n;
  bench.start();
  for (var i = 0; i < n; ++i) {
    require.resolve('os');
    require.resolve('../common.js');
  }
  bench.end(n);
}
```
and benchmark/module/require.js:
```js
'use strict';

var common = require('../common.js');

var bench = common.createBenchmark(main, {
  n: [512]
});

function main(opts) {
  var n = opts.n;
  bench.start();
  for (var i = 0; i < n; ++i) {
    require('os');
    require('../common.js');
  }
  bench.end(n);
}
```
bench result:
```shell
module/require.js
module/require.js n=512: 210.35436700941204

module/require.resolve.js
module/require.resolve.js n=512: 487.8754633465001
```

* bench in before add `resolve` method: 
  only test benchmark/module/require.js
  bench result:
```shell
module/require.js
module/require.js n=512: 229.30579491599653
```